### PR TITLE
Update command-line-args type to satisfy typical-usage-example of it

### PR DESCRIPTION
**Pull request by shibukawa**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/75lb/command-line-args/wiki/Typical-usage-example
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by 75lb — COMMENTED** [review]
the `description` and `typeLabel` properties are not related to command-line-args, they are used by a different package called command-line-usage. Yes, most people add those properties to a command-line-args `OptionDefinition` array type before passing it into command-line-usage but those two properties belong to, and are exclusively used by command-line-usage. I'm not a typescript expert but these two properties do not belong in command-line-args, i'm not sure how to handle this.

**Comment by shibukawa** [comment]
@75lb Thank you for your comment. Your are right. I found another solution to make the object works with command-line-args. New solution doesn't include property names of other package.

https://stackoverflow.com/questions/33836671/typescript-interface-that-allows-other-properties

Thanks!

File: types/command-line-args/index.d.ts [hunk 1]
```diff
@@ -84,6 +84,8 @@ declare namespace commandLineArgs {
          * One or more group names the option belongs to.
          */
         group?: string | string[];
+
+	[x: string]: any;
```
Comment (gabritto) [inline types/command-line-args/index.d.ts]: Indentation seems wrong here. Could you fix that?

**Review by gabritto — CHANGES_REQUESTED** [review]

**Comment by 75lb** [comment]
@shibukawa since we both agree this PR is not required, could you close it please

**Pull request closed without merging**